parasolr/query/queryset.py

```diff
@@ -532,7 +537,8 @@ def none(self) -> 'SolrQuerySet':
         qs_copy.search_qs = ['NOT *:*']
         return qs_copy
 
-    def _clone(self) -> 'SolrQuerySet':
+    # def _clone(self) -> 'SolrQuerySet':
```

Comment:
i think if you do `from __future__ import annotations` you can use `SolrQuerySet` as a return type inside its definition without getting `NameError`s - fyi

Reply:
I got an import error when I tried this; if I'm reading the docs correctly it was added in py3.7

Reply:
i think in some discussion i saw that the `typing` module mysteriously makes things way slower in 3.6, but that they improved it substantially in 3.7. might be worth having parasolr default to 3.7 instead

Reply:
We're using it on apps that are still python 3.5/3.6, I don't think we should do that until we get them upgraded — but would be great to drop older versions of python when we can.